File: frontend/src/api/summary.ts
import client from './client';

interface SummaryData {
  id: number;
  paper_id: number;
  summary_cn: string | null;
  key_points_cn: string | null;
  model_used: string;
  tokens_used: number;
  generated_at: string;
  error_message: string | null;
}

export async function summarizePaper(paperId: number): Promise<SummaryData> {
  const { data } = await client.post(`/summary/${paperId}`);
  return data;
}

export async function batchSummarize(limit = 10): Promise<{ success: number; failed: number; errors: { paper_id: number; error: string }[] }> {
  const { data } = await client.post('/summary/batch', { limit });
  return data;
}

File: backend/app/schemas/summary.py
import datetime

from pydantic import BaseModel, Field


class SummaryResponse(BaseModel):
    model_config = {"from_attributes": True, "protected_namespaces": ()}

    id: int
    paper_id: int
    summary_cn: str | None
    key_points_cn: str | None
    model_used: str | None
    tokens_used: int
    status: str = "completed"
    generated_at: datetime.datetime | None
    error_message: str | None
    source_type: str | None = None
    source_chars: int = 0


class BatchSummaryRequest(BaseModel):
    limit: int = Field(default=10, ge=1, le=50)


class BatchSummaryResponse(BaseModel):
    success: int
    failed: int
    errors: list[dict] = []


class SummaryStatsResponse(BaseModel):
    total_summaries: int
    pending_count: int
    failed_count: int
